Rock/Web/UI/RockPage.cs

```diff
@@ -831,6 +831,56 @@ protected override void OnInit( EventArgs e )
                     }
                 }
 
+                //Add Favicon to the header
+                #region favicon
+                HtmlLink favIcon16Sq = new HtmlLink();
+                HtmlLink favIcon32Sq = new HtmlLink();
+                HtmlLink favIcon144Sq = new HtmlLink();
+                HtmlLink favIcon180Sq = new HtmlLink();
+                HtmlLink favIcon192Sq = new HtmlLink();
+                HtmlLink favIcon = new HtmlLink();
+
+                favIcon16Sq.Attributes.Add( "rel", "apple-touch-icon-precomposed" );
+                favIcon16Sq.Attributes.Add( "sizes", "16x16" );
+                favIcon32Sq.Attributes.Add( "rel", "apple-touch-icon-precomposed" );
+                favIcon32Sq.Attributes.Add( "sizes", "32x32" );
+                favIcon144Sq.Attributes.Add( "rel", "apple-touch-icon-precomposed" );
+                favIcon144Sq.Attributes.Add( "sizes", "144x144" );
+                favIcon180Sq.Attributes.Add( "rel", "apple-touch-icon-precomposed" );
+                favIcon180Sq.Attributes.Add( "sizes", "180x180" );
+                favIcon192Sq.Attributes.Add( "rel", "apple-touch-icon-precomposed" );
+                favIcon192Sq.Attributes.Add( "sizes", "192*192" );
+                favIcon.Attributes.Add( "rel", "shortcut icon" );
+
+
+                if ( Site.FavIconBinaryFileId.HasValue )
+                {
+                    favIcon16Sq.Attributes.Add( "href", ResolveRockUrl( string.Format( "~/GetImage.ashx?id={0}&maxwidth={1}&maxheight={1}", Site.FavIconBinaryFileId.ToString(), "16" ) ) );
+                    favIcon32Sq.Attributes.Add( "href", ResolveRockUrl( string.Format( "~/GetImage.ashx?id={0}&maxwidth={1}&maxheight={1}", Site.FavIconBinaryFileId.ToString(), "32" ) ) );
+                    favIcon144Sq.Attributes.Add( "href", ResolveRockUrl( string.Format( "~/GetImage.ashx?id={0}&maxwidth={1}&maxheight={1}", Site.FavIconBinaryFileId.ToString(), "144" ) ) );
+                    favIcon180Sq.Attributes.Add( "href", ResolveRockUrl( string.Format( "~/GetImage.ashx?id={0}&maxwidth={1}&maxheight={1}", Site.FavIconBinaryFileId.ToString(), "180" ) ) );
+                    favIcon192Sq.Attributes.Add( "href", ResolveRockUrl( string.Format( "~/GetImage.ashx?id={0}&maxwidth={1}&maxheight={1}", Site.FavIconBinaryFileId.ToString(), "192" ) ) );
+                    favIcon.Attributes.Add( "href", ResolveRockUrl( string.Format( "~/GetImage.ashx?id={0}&maxwidth={1}&maxheight={1}", Site.FavIconBinaryFileId.ToString(), "192" ) ) );
+                }
+                else
+                {
+                    favIcon16Sq.Attributes.Add( "href", ResolveRockUrl( "~/Assets/Icons/touch-icon-iphone.png", true ) );
```

Comment:
@shivambareria Before adding the defaults, can you try to add code that checks to see if the <head> section already has a <link> element with a rel='shortcut icon', and if so not add the defaults. This is because while you removed the hard-coded icon references in our standard themes, customers will have other themes with the icons hard-coded in the layout, and we don't want to add them automatically in these cases.